tests/unit/parser/test_unit_tests.py

```diff
@@ -173,3 +226,62 @@ def test_multiple_unit_tests(self):
         for unit_test in self.parser.manifest.unit_tests.values():
             self.assertEqual(len(unit_test.depends_on.nodes), 1)
             self.assertEqual(unit_test.depends_on.nodes[0], "model.snowplow.my_model")
+
+    def _parametrize_test_promote_non_none_row(
+        self, unit_test_fixture_yml, fixture_expected_field_format, expected_rows
+    ):
+        block = self.yaml_block_for(unit_test_fixture_yml, "test_my_model.yml")
+
+        UnitTestParser(self.parser, block).parse()
+
+        self.assert_has_manifest_lengths(self.parser.manifest, nodes=1, unit_tests=1)
+        unit_test = list(self.parser.manifest.unit_tests.values())[0]
+        expected = UnitTestDefinition(
+            name="test_my_model_null_handling",
+            model="my_model",
+            resource_type=NodeType.Unit,
+            package_name="snowplow",
+            path=block.path.relative_path,
+            original_file_path=block.path.original_file_path,
+            unique_id="unit_test.snowplow.my_model.test_my_model_null_handling",
+            given=[],
+            expect=UnitTestOutputFixture(format=fixture_expected_field_format, rows=expected_rows),
+            description="unit test description",
+            overrides=None,
+            depends_on=DependsOn(nodes=["model.snowplow.my_model"]),
+            fqn=["snowplow", "my_model", "test_my_model_null_handling"],
+            config=UnitTestConfig(),
+            schema="test_schema",
+        )
+        expected.build_unit_test_checksum()
+        assertEqualNodes(unit_test, expected)
+
+    def test_expected_promote_non_none_row_dct(self):
+        expected_rows = [
+            {"id": 6, "col1": "f"},
+            {"id": None, "col1": "e"},
+            {"id": None, "col1": "d"},
+        ]
+        self._parametrize_test_promote_non_none_row(
+            UNIT_TEST_NONE_ROWS_SORT, UnitTestFormat.Dict, expected_rows
+        )
+
+    def test_expected_promote_non_none_row_csv(self):
+        expected_rows = [
+            {"id": "6", "col1": "f"},
+            {"id": None, "col1": "e"},
+            {"id": None, "col1": "d"},
+        ]
+        self._parametrize_test_promote_non_none_row(
+            UNIT_TEST_NONE_ROWS_SORT_CSV, UnitTestFormat.CSV, expected_rows
+        )
+
+    def test_expected_promote_non_none_row_sql(self):
+        expected_rows = "select null\n" + "select 1"
+        self._parametrize_test_promote_non_none_row(
+            UNIT_TEST_NONE_ROWS_SORT_SQL, UnitTestFormat.SQL, expected_rows
+        )
+
+    def test_no_full_row_does_not_raise_exception(self):
+        block = self.yaml_block_for(UNIT_TEST_NONE_ROWS_SORT_FAILS, "test_my_model.yml")
+        UnitTestParser(self.parser, block).parse()
```

Comment:
This test can more precisely test the expected event was fired with something like 
```py
from dbt_common.events.event_manager_client import add_callback_to_manager
from dbt_common.events.types import SystemStdErr
from tests.utils import EventCatcher
...

def test_no_full_row_does_not_raise_exception(self):
  catcher = EventCatcher(SystemStdErr)
  add_callback_to_manager(catcher.catch)
  ... 
  UnitTestParser(self.parser, block).parse()
  assert len(catcher.caught_events) == 1
```